Source/Public/Publish-Bicep.ps1

```diff
@@ -1,14 +1,30 @@
 function Publish-Bicep {
     [CmdletBinding()]
     param (
-        [Parameter(Mandatory, Position = 1)]
+        [Parameter(ParameterSetName = 'Default', Mandatory, Position = 1)]
+        [Parameter(ParameterSetName = 'Registry', Position = 1)]
         [ValidateNotNullOrEmpty()]
-        [ValidateScript({Test-Path $_})]
+        [ValidateScript( { Test-Path $_ })]
         [string]$Path,
 
-        [Parameter(Mandatory, Position = 2)]
+        [Parameter(ParameterSetName = 'Default', Position = 2)]
         [ValidateNotNullOrEmpty()]
-        [string]$Target          
+        [string]$Target,
+        
+        [Parameter(ParameterSetName = 'Registry', Mandatory, Position = 2)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Registry,
+
+        [Parameter(ParameterSetName = 'Registry', Mandatory, Position = 3)]
+        [ValidateNotNullOrEmpty()]
+        [ValidateScript( { $_ -notlike '/*' },
```

Comment:
While the $registry should be validated, this one is actually quite pointless. Its better to simply clean any trailing / in the code below, for example `$Repository = $Repository.TrimStart('/')` in the begin block. that way both variants works.